packages/amplify-graphql-api-construct-tests/src/sql-datatabase-controller.ts

```diff
@@ -81,110 +152,219 @@ export class SqlDatatabaseController {
+          this.options.password,
+          dbConfig.endpoint,
+          dbConfig.port,
+          this.options.dbname,
+        ),
+      });
+      const dbConnectionStringConfigMultiple = await storeDbConnectionStringConfig({
+        region: this.options.region,
+        pathPrefix,
+        connectionUri: [
+          'mysql://username:password@host:3306/dbname',
+          this.getConnectionUri(
+            engine,
+            this.options.username,
+            this.options.password,
+            dbConfig.endpoint,
+            dbConfig.port,
+            this.options.dbname,
+          ),
+        ],
+      });
+      const parameters = {
+        ...dbConnectionConfigSSM,
+        ...dbConnectionStringConfigSSM,
+        ...dbConnectionStringConfigMultiple,
+      };
+      if (!dbConnectionConfigSSM) {
+        throw new Error('Failed to store db connection config for SSM');
+      }
+      console.log(`Stored db connection config in SSM: ${JSON.stringify(Object.keys(parameters))}`);
+
+      this.databaseDetails = {
+        dbConfig: {
+          endpoint: dbConfig.endpoint,
+          port: dbConfig.port,
+          dbName: this.options.dbname,
+          strategyName: `${engine}DBStrategy`,
+          dbType: engine === 'postgres' ? 'POSTGRES' : 'MYSQL',
+          vpcConfig: extractVpcConfigFromDbInstance(dbConfig.dbInstance),
+        },
+        connectionConfigs: {
+          ssm: dbConnectionConfigSSM,
+          secretsManager: dbConnectionConfigSecretsManager,
+          secretsManagerCustomKey: dbConnectionConfigSecretsManagerCustomKey,
+          secretsManagerManagedSecret: {
+            databaseName: this.options.dbname,
+            hostname: dbConfig.endpoint,
+            port: dbConfig.port,
+            secretArn: dbConfig.secretArn,
+          },
+          connectionUri: dbConnectionStringConfigSSM,
+          connectionUriMultiple: dbConnectionStringConfigMultiple,
+        },
+      };
+      return this.databaseDetails;
+    } else {
+      // this.enableLocalTesting
       const secretArn = dbConfig.secretArn;
       const secretsManagerClient = new SecretsManagerClient({ region: this.options.region });
       const secretManagerCommand = new GetSecretValueCommand({
         SecretId: secretArn,
       });
       const secretsManagerResponse = await secretsManagerClient.send(secretManagerCommand);
-      const { password: managedPassword } = JSON.parse(secretsManagerResponse.SecretString);
-      if (!managedPassword) {
-        throw new Error('Unable to get RDS cluster master user password');
+      const secretArnWithCustomKey = secretsManagerResponse.ARN; // arn:aws:secretsmanager:us-west-2:637423428135:secret:yjLvhaVbVT-secret-custom-key-BTqGiz
+      const keyArn = 'arn:aws:kms:us-west-2:637423428135:key/a84fe5fe-ee01-44bf-b94e-57604a109c77'; // ELVIN - GET THE MASTER CREDENTIALS KMS KEY SOMEHOW (like "arn:aws:kms:us-west-2:637423428135:key/62ec591f-7188-49db-8819-87c15727908f")
```

Comment:
Let's scope this down to only handle DB managed secrets for now. Remove these hard-coded values.